l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,334 @@
+        uint256 _l2BatchNumber,
+        uint256 _l2MessageIndex,
+        uint16 _l2TxNumberInBatch,
+        bytes32[] calldata _merkleProof,
+        TxStatus _status
+    ) external view override returns (bool) {
+        address stateTransition = getStateTransition(_chainId);
+        return
+            IZkSyncStateTransition(stateTransition).proveL1ToL2TransactionStatus(
+                _l2TxHash,
+                _l2BatchNumber,
+                _l2MessageIndex,
+                _l2TxNumberInBatch,
+                _merkleProof,
+                _status
+            );
+    }
+
+    /// @notice forwards function call to Mailbox based on ChainId
+    function l2TransactionBaseCost(
+        uint256 _chainId,
+        uint256 _gasPrice,
+        uint256 _l2GasLimit,
+        uint256 _l2GasPerPubdataByteLimit
+    ) external view returns (uint256) {
+        address stateTransition = getStateTransition(_chainId);
+        return
+            IZkSyncStateTransition(stateTransition).l2TransactionBaseCost(
+                _gasPrice,
+                _l2GasLimit,
+                _l2GasPerPubdataByteLimit
+            );
+    }
+
+    /// @notice the mailbox is called directly after the sharedBridge received the deposit
+    /// this assumes that either ether is the base token or
+    /// the msg.sender has approved mintValue allowance for the sharedBridge.
+    /// This means this is not ideal for contract calls, as the contract would have to handle token allowance of the base Token
+    function requestL2TransactionDirect(
+        L2TransactionRequestDirect calldata _request
+    ) public payable override nonReentrant returns (bytes32 canonicalTxHash) {
+        uint256 l1GasPriceConverted;
+        {   
+            address token = baseToken[_request.chainId];
+            if (token == ETH_TOKEN_ADDRESS) {
+                require(msg.value == _request.mintValue, "Bridgehub: msg.value mismatch");
+                // kl todo it would be nice here to be able to deposit weth instead of eth
+                sharedBridge.bridgehubDepositBaseToken{value: _request.mintValue}(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+                l1GasPriceConverted = tx.gasprice;
+                require(_request.l1GasPriceConverted == 0, "Bridgehub: l1GasPriceConverted not 0 for ethBased chain");
+            } else {
+                require(msg.value == 0, "Bridgehub: non-eth bridge with msg.value");
+                // note we have to pass token, as a bridge might have multiple tokens.
+                sharedBridge.bridgehubDepositBaseToken(_request.chainId, msg.sender, token, _request.mintValue);
+
+                l1GasPriceConverted = _request.l1GasPriceConverted;
+                require(_request.l1GasPriceConverted >= tx.gasprice * baseTokenGasPriceMinMultiplier[token], "Bridgehub: gasPriceConverted not sufficient");
```

Comment:
I agree on the first point, I will move it to the STC. However, on the second point I don't understand, you say the ETH should not be a special case, but then you say 

>  I don't see any reason in allowing them to even multiply tx.gasprice in case ETH is native token

If I don't allow them to mulitply tx.gasprice then it is a special case, no? So should we allow a multiplier for Eth or not?

Reply:
Sorry, I meant "not allowing"